### PR TITLE
fix: added the roles type to PolicyRepresentation type,

### PR DESCRIPTION
**Pull request by fritzforsit**
enabling adding roles to policies.
Should also fix 
https://github.com/keycloak/keycloak-nodejs-admin-client/issues/402#issue-1122983563

Type conforms to the Keycloak docs:
https://www.keycloak.org/docs-api/13.0/javadocs/org/keycloak/representations/idm/authorization/RolePolicyRepresentation.RoleDefinition.html

### CODE REVIEW

**Review by vaibhavkumbhar1 — APPROVED** [review]

**Comment by vaibhavkumbhar1** [comment]
@fritzforsit  Workaround for create role

 await this.keycloakClient.clients.createPolicy({
                id:appIdentifier,
                realm: tenantId,
                type: ""

            },{
                name:roleEntity.roleId,
                type:"role",
                config:{
                    roles:"[{\"id\":\"b2057254-ac06-4224-8570-4ebfe766f0a0\",\"required\":true}]"
                }
            })

**Comment by jonkoops** [comment]
I don't see this field in the [current version of the documentation](https://www.keycloak.org/docs-api/16.1/rest-api/index.html#_policyrepresentation). Can you point me to where this should be present?

File: src/defs/policyRepresentation.ts [hunk 1]
```diff
@@ -31,4 +31,5 @@ export default interface PolicyRepresentation {
   scopes?: string[];
   type?: string;
   users?: string[];
+  roles?: {id: string, required?: boolean}[];
```
Comment (jonkoops) [inline src/defs/policyRepresentation.ts:34]: Can we split this type out into a separate interface? Where is this data structure documented/in the code?

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Comment by jonkoops** [comment]
@fritzforsit could you rebase this PR? Some changes have landed in `main` that should resolve the problems with the CI.

File: src/defs/policyRepresentation.ts (hunk 1 above)
Comment (fritzforsit) [reply src/defs/policyRepresentation.ts]: > Can we split this type out into a separate interface? Where is this data structure documented/in the code?

The roles Payload now has its own Interface and the rebase is done. Hopefully the CI now works!

**Comment by mastery-amit-sahani** [comment]
@jonkoops Could not find any reference in the API docs. But I guess you can see this in action in the below links.
https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/core/src/main/java/org/keycloak/representations/idm/authorization/RolePolicyRepresentation.java#L27
https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/permission/UMAPolicyProviderFactory.java#L434

**Pull request merged**